modules/autocomplete-picker/main/index.coffee

```diff
@@ -0,0 +1,221 @@
+          ddClass: 'hx-autocomplete-picker-dropdown'
+      })
+
+      @_.renderer = resolvedOptions.renderer or menu.renderer()
+      @_.menu = menu
+      menu.renderer(renderWrapper)
+
+      noResultsItem =
+        text: resolvedOptions.noResultsText
+        unselectable: true
+
+      loadingItem =
+        text: resolvedOptions.loadingText
+        unselectable: true
+
+      otherResultsItem =
+        text: resolvedOptions.otherResultsText
+        unselectable: true
+        heading: true
+
+      renderMenu = (items) ->
+        menu.items(items)
+        menu.dropdown._.setupDropdown(menu.dropdown._.dropdown.node())
+
+      populateMenu = (term) ->
+        renderMenu([loadingItem])
+        feed.filter term, (results, otherResults) ->
+          if results.length is 0
+            results.push(noResultsItem)
+          if otherResults.length > 0
+            otherResults = [otherResultsItem].concat(otherResults)
+          renderMenu(results.concat(otherResults))
+
+      debouncedPopulate = hx.debounce debounceDuration, populateMenu
+
+      setValue = (item) =>
+        setPickerValue(this, [item], 'user')
+        menu.hide()
+
+      input = hx.detached('input').class('hx-autocomplete-picker-input')
+        .on 'input', (e) -> debouncedPopulate(e.target.value)
+        .on 'keydown', (e) ->
+          if input.value().length
+            if (e.which or e.keyCode) is enterKeyCode and menu.cursorPos is -1
+              topItem = menu.items()[0]
+              if not topItem.unselectable
+                setValue(topItem)
+
+      menu.dropdown.on 'showstart', ->
+        input.value('')
+        menu.dropdown._.dropdown.prepend(input)
+        debouncedPopulate(input.value())
+
+      menu.dropdown.on 'showend', ->
+        input.node().focus()
+
+      menu.on 'change', (item) =>
+        if item? and item.content?
+          setValue(item.content)
+
+      menu.pipe this, '', ['highlight']
+      menu.dropdown.pipe this, 'dropdown'
+
+      if resolvedOptions.value
+        @value(resolvedOptions.value)
+      else
+        valueText.text(resolvedOptions.chooseValueText)
+
+      if resolvedOptions.disabled
+        @disabled(resolvedOptions.disabled)
+
+
+  clearCache: ->
+    @_.feed.clearCache()
+    this
+
+  hide: ->
+    @_.menu.hide()
+    this
+
+  disabled: (disable) ->
+    menuDisable = @_.menu.disabled(disable)
+    # menu.disabled returns the wrong 'this' so we return the correct things below
+    if disable? then this
+    else menuDisable
+
+  items: (items) ->
+    if arguments.length
+      if validateItems(@_.feed, items)
+        @_.feed.items(items)
+        @value(@_.current)
+      this
+    else
+      @_.feed.items()
+
+  value: (value, callback) ->
+    _ = @_
+    if arguments.length
+      _.valueText.text(_.options.loadingText)
+      _.feed.filter _.valueLookup(value), (results) =>
+        setPickerValue(this, results, 'api')
+        callback?(value, results.length is 1)
```

Comment:
I realised that this isn't matching up with what setPickerValue is doing - It should return the first result if the results length is > 0, not the value passed in.